pkg/virt-launcher/virtwrap/network/podnic_test.go

```diff
@@ -80,22 +82,29 @@ var _ = Describe("podNIC", func() {
 			Expect(ok).To(BeTrue(), "SetupPhase1 should return an error of type CriticalNetworkError")
 		})
 	})
-	PContext("when DHCP startup fails", func() {
+	Context("when DHCP startup fails", func() {
+		var (
+			tmpDir string
+		)
 		BeforeEach(func() {
+			var err error
+			tmpDir, err = ioutil.TempDir(os.TempDir(), "dhcp")
+			Expect(err).ToNot(HaveOccurred())
 			mockNetwork.EXPECT().StartDHCP(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(errors.New("podnic_test: forcing failure at DHCP start"))
 		})
+		AfterEach(func() {
+			os.RemoveAll(tmpDir)
+		})
 		It("phase2 should panic", func() {
-			testDhcpPanic := func() {
-				domain := NewDomainWithBridgeInterface()
-				vmi := newVMIBridgeInterface("testnamespace", "testVmName")
-				api.NewDefaulter(runtime.GOARCH).SetObjectDefaults_Domain(domain)
-				podnic, err := newPodNICWithMocks(vmi)
-				Expect(err).ToNot(HaveOccurred())
-				podnic.storeCachedDomainIface(domain.Spec.Devices.Interfaces[0])
-				podnic.dhcpConfigurator.ExportConfiguration(cache.DHCPConfig{Name: "eth0"})
-				Expect(podnic.PlugPhase2(domain)).To(Succeed())
-			}
-			Expect(testDhcpPanic).To(Panic())
+			domain := NewDomainWithBridgeInterface()
+			vmi := newVMIBridgeInterface("testnamespace", "testVmName")
+			api.NewDefaulter(runtime.GOARCH).SetObjectDefaults_Domain(domain)
+			podnic, err := newPodNICWithMocks(vmi)
+			podnic.dhcpConfigurator.SetDHCPStartedDIrectory(tmpDir)
+			Expect(err).ToNot(HaveOccurred())
+			podnic.storeCachedDomainIface(domain.Spec.Devices.Interfaces[0])
+			podnic.dhcpConfigurator.ExportConfiguration(cache.DHCPConfig{Name: "eth0"})
+			Expect(func() { podnic.PlugPhase2(domain) }).To(Panic())
```

Comment:
If you want to remove the encapsulating function, I think it becomes more readable / focused if you move all but the `Expect(func() { podnic.PlugPhase2(domain) }).To(Panic())` expectation to a `BeforeEach` block, since that's just setting up the test. 

Yes... I'm aware you'd have to set up the podnic as a variable. That's good.